src/App.tsx

```diff
@@ -1,27 +1,108 @@
+import React, { useState } from 'react';
+import { Todo } from './types';
 import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import todosData from './api/todos';
+import usersData from './api/users';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const users = usersData;
+
+  const initialTodos: Todo[] = todosData.map(todo => {
+    const user = users.find(user => user.id === todo.userId);
+    return {
+      ...todo,
+      user: user || { id: 0, name: 'Unknown', email: '' },
+    };
+  });
+  
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState<string>('');
+  const [userId, setUserId] = useState<number>(0);
+  const [error, setError] = useState<{ title: string; user: string }>({ title: '', user: '' });
+
+  const handleAddTodo = (e: React.FormEvent) => {
+    e.preventDefault();
+    setError({ title: '', user: '' });
+
+    const newErrors = { title: '', user: '' };
+
+    if (!title) {
+      newErrors.title = 'Please enter a title';
+    }
+
+    if (userId === 0) {
+      newErrors.user = 'Please choose a user';
+    }
+
+    if (newErrors.title || newErrors.user) {
+      setError(newErrors);
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...todos.map(todo => todo.id), 0) + 1,
```

Comment:
The logic for generating a new todo ID using `Math.max(...todos.map(todo => todo.id), 0) + 1` assumes that there is at least one todo in the `todos` array. If the array is empty, this will correctly start IDs from 1, but ensure that this logic aligns with your application's requirements for ID generation.